### PR TITLE
Зорин - Лабораторная работа №3

### PR DESCRIPTION
**Pull request by oleg-w570**

### CODE REVIEW

File: modules/binary-tree/include/binary_tree.h [hunk 1]
```diff
@@ -3,43 +3,42 @@
...
-  explicit BinaryTree(const std::vector<int>& v)
-      : root(nullptr), curr(nullptr) {
-     for (const auto& value : v) {
-         Insert(value);
-     }
-  }
-
-  Node* Find(int k) const;
+  explicit BinaryTree(int v);
+  explicit BinaryTree(const std::vector<int> &v);
```
Comment (dkurt) [inline modules/binary-tree/include/binary_tree.h:30]: Why so many changes in origin code?

File: modules/binary-tree/include/binary_tree.h [hunk 2]
```diff
@@ -3,43 +3,42 @@
...
-  Node* root;
-  Node* curr;
-  Node* FindMin(Node* node) const;
-  Node* FindMax(Node* node) const;
-  Node* FindNext(Node* node) const;
+  Node *root;
+  Node *curr;
+  static Node *FindMin(Node *node);
+  static Node *FindMax(Node *node);
+  static Node *FindNext(Node *node);
```
Comment (dkurt) [inline modules/binary-tree/include/binary_tree.h:26]: Let's minimize changes in the base module. Keep only that required for the app.

Comment (oleg-w570) [reply modules/binary-tree/include/binary_tree.h]: Сделал меньше изменений

**Comment by TupoaMudille** [comment]
@oleg-w570 please, resolve conflict. You can create new pr with update topic and delete topic from this

**Comment by oleg-w570** [comment]
@TupoaMudille вроде исправил

**Pull request merged**